### DISCUSSION TITLE
What are the remaining iptables rules for?

### DISCUSSION THREAD

**Discussion by ttc0419** [cilium/cilium#36042] [Q&A] [answered]
According to https://docs.cilium.io/en/stable/network/ebpf/iptables/, there're still some iptables rules used by cilium, even the replace kube-proxy option is set.  What are they and how are they used. Is there a way to disable iptables usage entirely like the setup in https://www.youtube.com/watch?v=bIRwSIwNHC0&t=634s?

**Comment by joestringer** [comment, marked as answer]
There are three capabilities described here implemented with a few static iptables rules: https://github.com/cilium/cilium/issues/12879 .

Additionally, as long as you have iptables enabled, it is possible for other rules installed by different software components to affect Cilium's forwarding behavior. To ensure that Cilium's features still work correctly in the presence of those other rules, Cilium also creates other rules which are intended to make sure that Cilium integrates correctly with those other components, for instance by ensuring that traffic managed by Cilium Layer 7 network policies is directed towards Cilium-managed proxies for implementing the network policy.
